python/activity_stream/activity_stream.py

```diff
@@ -711,16 +725,30 @@ def note_update(self,entity, note_id):
         attachment_metadata["created_at"] = time.mktime(dtime.timetuple())
  
         note_thread_data.append(attachment_metadata)
+        current_attachments.append(attachment_metadata)
         self._data_manager.db_insert_note_update(note_id, note_thread_data)
 
         ids_to_process = self._data_manager.load_activity_data(entity["type"], 
                                                                entity["id"],
                                                                self.MAX_STREAM_LENGTH)     
         activity_id = ids_to_process[-1]
 
-        for widget in self._activity_stream_data_widgets.values():
-            if isinstance(widget, NoteWidget) and widget.note_id == note_id:                                               
-                widget._add_attachment_group([attachment_metadata], True)
+        for widget in self._activity_stream_data_widgets.values():           
+            if isinstance(widget, NoteWidget) and widget.note_id == note_id:     
+                
+                # Force a deletion to the attachment groups to refresh everything in the same group and same line
+                for x in widget._attachment_group_widgets.values():
+                    # remove widget from layout:
+                    widget.ui.reply_layout.removeWidget(x)
```

Comment:
Rather than access the underlying ui structure, the NoteWidget should expost the reply_layout as a property. We want to be able to change the UI layout of the NoteWidget without breaking higher-level widgets like would happen here.